src/test/java/pl/marcin/project/repository/CustomerRepositoryListBasedTest.java

```diff
@@ -38,7 +40,166 @@ public void shouldReturnCustomerWithId2() {
 
         //then
         Assertions.assertEquals(id2, customer.getId());
+    }
+
+    @Test
+    public void shouldThrowExceptionIfNotFindCustomerById() {
+        //given
+        int id = 1;
+        int id2 = 2;
+        //when:
+        Customer customer2 = Customer.builder().id(id2).build();
+        customerRepository.saveCustomer(customer2);
+
+        //then:
+        assertThrows(RuntimeException.class, () -> {
+            customerRepository.findCustomer(id);
+        });
+    }
+
+    @Test
+    public void shouldThrowExceptionMessageNoSuchCustomerById() {
+        //given
+        int id = 1;
+
+        //when
+        Throwable exception = assertThrows(RuntimeException.class, () -> {
+            customerRepository.findCustomer(id);
+        });
+
+        //then
+        Assertions.assertEquals("There is no such customer", exception.getMessage());
+    }
+
+    @Test
+    public void shouldReturnListOfAllCustomers() {
+        //given
+        int id = 1;
+        int id2 = 2;
+        Customer customer1 = Customer.builder().id(id).build();
+        Customer customer2 = Customer.builder().id(id2).build();
+        customerRepository.saveCustomer(customer1);
+        customerRepository.saveCustomer(customer2);
+        //when
+        List<Customer> list = customerRepository.getAllCustomers();
+        //then
+        Assertions.assertEquals(List.of(customer1, customer2), list);
+    }
+
+    @Test
+    public void shouldReturnTrueIfDeleted() {
+        //given
+        int id = 1;
+        Customer customer1 = Customer.builder().id(id).build();
+        customerRepository.saveCustomer(customer1);
+        //when
+        boolean ifDeleted = customerRepository.deleteCustomer(1);
+        //then
+        Assertions.assertTrue(ifDeleted);
+    }
+
+    @Test
+    public void shouldReturnFalseIfNotDeleted() {
+        //given
+        int id = 1;
+        Customer customer1 = Customer.builder().id(id).build();
+        customerRepository.saveCustomer(customer1);
+        //when
+        boolean ifDeleted = customerRepository.deleteCustomer(2);
+        //then
+        Assertions.assertFalse(ifDeleted);
+    }
+
+    @Test
+    public void shouldReturnCustomerId() {
+        //given
+        int id = 1;
+        Customer customer1 = Customer.builder()
+                .id(id)
+                .name("Jan")
+                .surname("Kowalski")
+                .address("Kraków")
+                .build();
+        //when
+        int returnedId = customerRepository.saveCustomer(customer1);
+        //then
+        Assertions.assertEquals(id, returnedId);
+    }
+
+    @Test
+    public void shouldEqualToSavedCustomer() {
```

Comment:
this test with `shouldReturnCustomerId` can be merged

Reply:
ok